test-network-function/lifecycle/suite.go

```diff
@@ -82,90 +81,162 @@ var drainTimeout = time.Duration(drainTimeoutMinutes) * time.Minute
 // All actual test code belongs below here.  Utilities belong above.
 //
 var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
+	configData := common.ConfigurationData{}
+	ginkgo.BeforeSuite(func() {
+		common.Loadconfiguration(&configData)
+		log.Info(configData.ContainersUnderTest)
+	})
+	ginkgo.BeforeEach(func() {
+		common.ReloadConfiguration(&configData)
+	})
 	if testcases.IsInFocus(ginkgoconfig.GinkgoConfig.FocusStrings, common.LifecycleTestKey) {
-		config := common.GetTestConfiguration()
-		log.Infof("Test Configuration: %s", config)
-
-		containersUnderTest := common.CreateContainersUnderTest(config)
-
-		log.Info(containersUnderTest)
-
-		for _, containerUnderTest := range containersUnderTest {
-			testNodeSelector(common.GetContext(), containerUnderTest.Oc.GetPodName(), containerUnderTest.Oc.GetPodNamespace())
-		}
 
-		for _, containerUnderTest := range containersUnderTest {
-			testGracePeriod(common.GetContext(), containerUnderTest.Oc.GetPodName(), containerUnderTest.Oc.GetPodNamespace())
-		}
+		testNodeSelector(&configData)
 
-		for _, containerUnderTest := range containersUnderTest {
-			testShutdown(containerUnderTest.Oc.GetPodNamespace(), containerUnderTest.Oc.GetPodName())
-		}
+		testGracePeriod(&configData)
 
-		for _, containersUnderTest := range containersUnderTest {
-			testDeployments(containersUnderTest.Oc.GetPodNamespace())
-		}
+		testShutdown(&configData)
 
-		for _, containerUnderTest := range containersUnderTest {
-			testOwner(containerUnderTest.Oc.GetPodNamespace(), containerUnderTest.Oc.GetPodName())
-		}
+		testPodAntiAffinity(&configData)
 
-		for _, containerUnderTest := range containersUnderTest {
-			testPodAntiAffinity(containerUnderTest.Oc.GetPodNamespace())
+		testPodsRecreation(&configData)
```

Comment:
Check the non intrusive only flag

Reply:
This test can run safely now even in our local machine. That's why I removed the check. I'll add it back

Reply:
fixed